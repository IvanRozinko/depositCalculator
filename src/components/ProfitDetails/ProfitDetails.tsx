import React, { useState } from 'react'
import { 
  Box, Collapse, Typography, makeStyles
} from '@material-ui/core';
import ArrowDropDownCircleIcon from '@material-ui/icons/ArrowDropDownCircleOutlined';
import InfoRow from '../InfoRow/InfoRow';

interface Props {
  pureProfit: string,
  purePercent: number
}

function ProfitDetails({ pureProfit, purePercent }: Props) {
  const [collapse, setCollapse] = useState(false);
  const [duration, setDuration] = useState('1ms')

  const useStyle = makeStyles({
    container: {
      marginBottom: 10
    },
    header: {
      '&:hover': {
        cursor: 'pointer'
      }
    },
    bold: {
      fontWeight: 600
    },
    "@keyframes rotate": {
      "0%": {
        transform: `rotateZ(${collapse ? '0' : '180deg'})`
      },
      "100%": {
        transform: `rotateZ(${collapse ? '180deg' : '360deg'})`
      }
    },
    arrow: {
      fontSize: 20,
      marginRight: 8,
      transform: `rotateZ(${collapse ? '180deg' : '0'})`,
      animation: `$rotate ${duration} ease-in-out`
    }
  })

  const styles = useStyle();

  const handleClick = () => {
    setDuration('300ms');
    setCollapse(!collapse);
  }

  return (
    <Box  className={styles.container}>
      <Box
        onClick={handleClick}
        display='flex'
        alignItems='center'
        className={styles.header}
      >
        <ArrowDropDownCircleIcon color='primary' className={styles.arrow} />
        <Typography className={styles.bold}>
          Після сплати податку
        </Typography>
      </Box>
      <Collapse in={collapse}>
        <InfoRow label='Проценти (дохід)' value={pureProfit} />
        <InfoRow label='Процентна ставка' value={`${purePercent} % на рік`} />
      </Collapse>
    </Box>  
  )
}

export default ProfitDetails
