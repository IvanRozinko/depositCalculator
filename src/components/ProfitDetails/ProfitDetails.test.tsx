import ProfitDetails from "./ProfitDetails";
import { shallow } from 'enzyme';
import { Box, Collapse } from '@material-ui/core';

describe('<ProfitDetails /> :', () => {
  const pureProfit = '100';
  const purePercent = 6.7;
  const wrapper = shallow(
    <ProfitDetails
      purePercent={purePercent}
      pureProfit={pureProfit}
    />
  )

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should not be expanded', () => {
    expect(wrapper.find(Collapse).props().in).toBe(false)
  })

  it('should be expanded on click', () => {
   wrapper.find(Box).at(1).simulate('click');
   expect(wrapper.find(Collapse).props().in).toBe(true);
  })
  
  it('should has 2 Inforow', () => {
    expect(wrapper.find('InfoRow').length).toBe(2)
  })
})