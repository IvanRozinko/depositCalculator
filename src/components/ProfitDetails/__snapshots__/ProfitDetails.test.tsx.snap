// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfitDetails /> : should render correctly 1`] = `
<Styled(MuiBox)
  className="makeStyles-container-1"
>
  <Styled(MuiBox)
    alignItems="center"
    className="makeStyles-header-2"
    display="flex"
    onClick={[Function]}
  >
    <Memo(ForwardRef(ArrowDropDownCircleOutlinedIcon))
      className="makeStyles-arrow-5"
      color="primary"
    />
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-bold-3"
    >
      Після сплати податку
    </WithStyles(ForwardRef(Typography))>
  </Styled(MuiBox)>
  <WithStyles(ForwardRef(Collapse))
    in={false}
  >
    <InfoRow
      label="Проценти (дохід)"
      value="100"
    />
    <InfoRow
      label="Процентна ставка"
      value="6.7 % на рік"
    />
  </WithStyles(ForwardRef(Collapse))>
</Styled(MuiBox)>
`;

exports[`<ProfitDetatails /> : should render correctly 1`] = `
<Styled(MuiBox)
  className="makeStyles-container-1"
>
  <Styled(MuiBox)
    alignItems="center"
    className="makeStyles-header-2"
    display="flex"
    onClick={[Function]}
  >
    <Memo(ForwardRef(ArrowDropDownCircleOutlinedIcon))
      className="makeStyles-arrow-5"
      color="primary"
    />
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-bold-3"
    >
      Після сплати податку
    </WithStyles(ForwardRef(Typography))>
  </Styled(MuiBox)>
  <WithStyles(ForwardRef(Collapse))
    in={false}
  >
    <InfoRow
      label="Проценти (дохід)"
      value="100"
    />
    <InfoRow
      label="Процентна ставка"
      value="6.7 % на рік"
    />
  </WithStyles(ForwardRef(Collapse))>
</Styled(MuiBox)>
`;
